### ISSUE TITLE
CA2201 With String.Format for Message Causes VBC Compilation Error

### ISSUE THREAD

**Issue by adraut** [dotnet/roslyn-analyzers#2702]
#### Analyzer package
[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version
2.9.3

#### Diagnostic ID
[CA2201](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2201-do-not-raise-reserved-exception-types?view=vs-2019)

#### Repro steps
VS 2019 16.1.6
Microsoft (R) Visual Basic Compiler version 3.100.119.28106 (58a4b1e7)

##### CA2201WithStringFormat.vbproj
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <RootNamespace>VbcStringError</RootNamespace>
    <TargetFramework>net472</TargetFramework>
    <OptionStrict>On</OptionStrict>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.3" />
  </ItemGroup>

</Project>
```

##### Class1.vb
```
Public Class Class1

    Public Sub A()
        Throw New Exception(String.Format("There is an error: {0}", NameOf(A)))
    End Sub

End Class
```
#### Expected behavior
CA2201 is raised.

#### Actual behavior
VBC compilation error
```
Severity = Error
Code = 
Description = {0}", NameOf(A)))
Project = CA2201WithStringFormat
File = VBC
Line = 1
Source = Build
```

It appears that is specific to the colon after the word `error`.  If the colon is removed or replaced, such as with an equals sign, the behavior is as expected.

I added a unit test to the `DoNotRaiseReservedExceptionTypesTests` unit test class like the one below and it passed.
```
        [Fact]
        public void CreateSystemExceptionWithMessage()
        {
            VerifyBasic(@"
Imports System

Namespace TestNamespace
	Class TestClass
		Private Shared Sub TestMethod()
            Throw New Exception(String.Format(""There is an error: {0}"", NameOf(TestMethod)))
        End Sub
	End Class
End Namespace",
            GetTooGenericBasicResultAt(7, 23, "System.Exception"));
        }
```

**Comment by mavasani** [comment]
@adraut Thanks for the report. Apparently this looks to be a bug in the VB Compiler. I have ported this to https://github.com/dotnet/roslyn/issues/37508
